src/getVendors.ts

```diff
@@ -0,0 +1,34 @@
+import { gql } from 'graphql-request';
+import client from './graphClient';
+import { Vendor } from './generated/graphql';
+
+interface VendorsQueryResponse {
+    vendors: Vendor[]
+}
+
+export const getVendorsQuery = gql`
+    query getVendors{
+        vendors {
+            id
+            title
+            website
+            logo
+            streetAddress
+            city
+            state
+            postalCode
+            isActive
+        }
+    }
+`;
+
+const getVendors = async (): Promise<Vendor[]> => {
+    try {
+        const data: VendorsQueryResponse = await client.request(getVendorsQuery);
+        return data.vendors;
+    } catch (e: unknown) {
```

Comment:
Wouldn't `e` be of type `Exception` ? 

(I'm genuinely asking since I dunno the TS/JS library)

Reply:
Typescript is weird with catch clauses. You would think it'd be type `Errror` (don't think there is Exception). Apparently it has to be `any` or `unknown` 

<img width="809" alt="Screen Shot 2022-03-06 at 9 13 43 AM" src="https://user-images.githubusercontent.com/27029446/156934100-1d865206-3a0e-470e-97db-e30c59b4ddd5.png">
e